utils/databake/src/primitives.rs

```diff
@@ -99,32 +137,65 @@ fn slice() {
     // Cannot use test_bake! as it's not possible to write a closed slice expression (&[1] has type &[usize; 1])
     let slice: &[bool] = &[];
     assert_eq!(Bake::bake(&slice, &Default::default()).to_string(), "& []");
+    assert_eq!(BakeSize::borrows_size(&slice), 0);
     let slice: &[bool] = &[true];
     assert_eq!(
         Bake::bake(&slice, &Default::default()).to_string(),
         "& [true]",
     );
+    assert_eq!(BakeSize::borrows_size(&slice), 1);
     let slice: &[bool] = &[true, false];
     assert_eq!(
         Bake::bake(&slice, &Default::default()).to_string(),
         "& [true , false]",
     );
+    assert_eq!(BakeSize::borrows_size(&slice), 2);
+    let slice: &[u8] = b"hello";
+    assert_eq!(
+        Bake::bake(&slice, &Default::default()).to_string(),
+        r#"b"hello""#,
+    );
+    assert_eq!(BakeSize::borrows_size(&slice), 5);
 }
 
 impl<T, const N: usize> Bake for [T; N]
 where
     T: Bake,
 {
     fn bake(&self, ctx: &CrateEnv) -> TokenStream {
-        self.as_slice().bake(ctx)
+        if core::mem::size_of::<T>() == core::mem::size_of::<u8>()
+            && core::any::type_name::<T>() == core::any::type_name::<u8>()
+        {
+            // Safety: self.as_ptr()'s allocation is at least self.len() bytes long,
+            // initialised, and well-alligned.
+            let bytestr =
+                byte_string(unsafe { core::slice::from_raw_parts(self.as_ptr() as *const u8, N) });
+            return quote!(*#bytestr);
+        }
```

Comment:
Suggestion: can you avoid the unsafe here by downcasting through Any?

Reply:
Is this the only open issue? Because this is pre-existing code

Reply:
and it doesn't work because it's a fat pointer

Reply:
(not for this PR but I do want to experiment with reducing this unsafe)